File: legacy/python-backend/app/services/strategy_service.py
from sqlalchemy.ext.asyncio import AsyncSession

from app.repositories.strategy_repository import (
    StrategyRepository
)


class StrategyService:

    @staticmethod
    async def create_strategy(
        db: AsyncSession,
        payload: dict
    ):

        return await StrategyRepository.create(
            db,
            payload
        )

    @staticmethod
    async def list_strategies(
        db: AsyncSession
    ):

        return await StrategyRepository.get_all(db)

    @staticmethod
    async def update_strategy(
        db: AsyncSession,
        strategy_id: str,
        update_data: dict
    ):
        strategy = await StrategyRepository.get_by_id(db, strategy_id)
        if not strategy:
            return None
        return await StrategyRepository.update(db, strategy, update_data)

    @staticmethod
    async def delete_strategy(
        db: AsyncSession,
        strategy_id: str
    ):
        strategy = await StrategyRepository.get_by_id(db, strategy_id)
        if not strategy:
            return False
        return await StrategyRepository.delete(db, strategy)

File: backend/src/services/strategyService.ts
import { randomUUID } from "crypto";
import { StrategyRepository } from "../repositories/strategyRepository";

export class StrategyService {
  static async createStrategy(payload: {
    name: string;
    description?: string | null;
    config: any;
    is_active?: boolean;
  }) {
    return StrategyRepository.create({
      id: randomUUID(),
      name: payload.name,
      description: payload.description,
      config: payload.config,
      is_active: payload.is_active !== undefined ? payload.is_active : true,
    });
  }

  static async listStrategies() {
    return StrategyRepository.getAll();
  }

  static async updateStrategy(strategyId: string, updateData: any) {
    const strategy = await StrategyRepository.getById(strategyId);
    if (!strategy) {
      return null;
    }
    return StrategyRepository.update(strategyId, updateData);
  }

  static async deleteStrategy(strategyId: string) {
    const strategy = await StrategyRepository.getById(strategyId);
    if (!strategy) {
      return false;
    }
    return StrategyRepository.delete(strategyId);
  }
}
export default StrategyService;
